opentasks-provider/src/main/java/org/dmfs/provider/tasks/processors/instances/Validating.java

```diff
@@ -43,50 +69,79 @@ public Validating(EntityProcessor<InstanceAdapter> delegate)
     public InstanceAdapter insert(SQLiteDatabase db, InstanceAdapter entityAdapter, boolean isSyncAdapter)
     {
         validate(entityAdapter, isSyncAdapter);
-        return mDelegate.insert(db, entityAdapter, isSyncAdapter);
+
+        Optional<Long> instanceId = new NullSafe<>(entityAdapter.taskAdapter().valueOf(TaskAdapter.ORIGINAL_INSTANCE_ID));
+        if (instanceId.isPresent())
+        {
+            String timeStampString = Long.toString(entityAdapter.taskAdapter().valueOf(TaskAdapter.ORIGINAL_INSTANCE_TIME).getTimestamp());
```

Comment:
Wouldn't it be better to read ORIGINAL_INSTANCE_TIME here as `Optional` as well and throw explicitly in case of absent?